openmsupply/open-msupply
# 1326 arv section update

Fixes #1326 

# 👩🏻‍💻 What does this PR do? 
- Add a conditional filter for the Select component. (as needed for the issue)
- Fix custom errors

# 🧪 How has/should this change been tested? 

- Select ARV status and reason
- Change status and check that reason list is updated
- If an invalid reason for the current status is selected an error is shown
- If you have an invalid reason for the CONTINUE status there are no other options in the list and you have to clear the field to make the error go a await, is this intuitive enough?
